hera_opm/mf_tools.py

```diff
@@ -1428,47 +1433,55 @@ def build_lstbin_makeflow_from_config(
         print("export BATCH_OPTIONS = {}".format(batch_options), file=f)
 
-        # get data files
-        datafiles = get_config_entry(config, "LSTBIN_OPTS", "data_files", required=True)
-        # encapsulate in double quotes
-        datafiles = [
-            "'{}'".format(
-                '"{}"'.format(os.path.join(parent_dir, df.strip('"').strip("'")))
-            )
-            for df in datafiles
-        ]
-
-        # get number of output files
-        if parallelize:
-            # get LST-specific config options
-            dlst = get_config_entry(config, "LSTBIN_OPTS", "dlst", required=True)
-            if dlst == "None":
-                dlst = None
-            else:
-                dlst = float(dlst)
-            lst_start = float(
-                get_config_entry(config, "LSTBIN_OPTS", "lst_start", required=True)
-            )
-            ntimes_per_file = int(
-                get_config_entry(
-                    config, "LSTBIN_OPTS", "ntimes_per_file", required=True
-                )
-            )
+        datafiles = get_lstbin_datafiles(config, parent_dir)
 
-            # pre-process files to determine the number of output files
-            _datafiles = [
-                sorted(glob.glob(df.strip("'").strip('"'))) for df in datafiles
-            ]
+        print("Searching for files in the following globs: ")
+        for df in datafiles:
+            print("  " + df.strip("'").strip('"'))
 
-            output = lstbin.config_lst_bin_files(
-                _datafiles,
-                dlst=dlst,
-                lst_start=lst_start,
-                ntimes_per_file=ntimes_per_file,
-            )
-            nfiles = len(output[2])
+        # pre-process files to determine the number of output files
+        _datafiles = [sorted(glob.glob(df.strip("'").strip('"'))) for df in datafiles]
+        _datafiles = [df for df in _datafiles if len(df) > 0]
+
+        def get(key: str, **kw):
+            if "default" in kw:
+                kw["required"] = False
+
+            return get_config_entry(config, "LSTBIN_OPTS", key, **kw)
+
+        if "outdir" in kwargs:
+            outdir = Path(kwargs["outdir"])
         else:
-            nfiles = 1
+            outdir = Path(get("outdir"))
+
+        lstbin_config_file = Path(outdir) / "file-config.yaml"
+
+        # Get dlst. Updated version supports leaving dlst unspecified or set as null.
+        # To support older versions which required string 'None', set that to None here.
+        dlst = get("dlst", default=None)
+        if dlst.lower() == "none":
+            warnings.warn(
+                "dlst should not be set to (string) 'None', but rather left unspecified in your TOML.",
+                DeprecationWarning,
+            )
+            dlst = None
+
+        file_config = lstbin_simple.make_lst_bin_config_file(
+            config_file=lstbin_config_file,
+            data_files=_datafiles,
+            clobber=get("overwrite", default=False),
```

Comment:
I'm also not a huge fan of implicitly embedding function calls as the keywords of kwargs, as it's not clear exactly what value is being passed in as the argument (in case something goes wrong and debugging is required). I think I'd prefer these option-fetching operations to happen outside of the function call, and using `get_config_entry` directly.

Reply:
Can do.